database_build/db_build.sql

```diff
@@ -3,29 +3,52 @@ BEGIN;
 DROP TABLE IF EXISTS memories, users, media CASCADE;
 
 CREATE TABLE memories (
-  id          SERIAL        PRIMARY KEY,
+  id          SERIAL          PRIMARY KEY,
+  user_id     SERIAL          NOT NULL,
   heading     VARCHAR(50)     NOT NULL,
   likes       INTEGER,
-  rating      INTEGER,
+  avgRating   DECIMAL,
   visits      INTEGER,
+  tag         VARCHAR(50)     NOT NULL,
   memory      VARCHAR(500)    NOT NULL
 );
 
 CREATE TABLE users (
-  id          SERIAL        PRIMARY KEY,
-  username    VARCHAR(64)   UNIQUE NOT NULL,
-  password    VARCHAR(64)  NOT NULL,
-  email    VARCHAR(64)  NOT NULL
+  id          SERIAL          PRIMARY KEY,
+  username    VARCHAR(64)     UNIQUE NOT NULL,
+  password    VARCHAR(64)     NOT NULL,
+  email       VARCHAR(64)     NOT NULL
 );
 
 CREATE TABLE media (
-  text_only    TEXT[],
-  audio    TEXT[],
-  video    TEXT[],
-  image    TEXT[]
+  text_only   INTEGER[],
```

Comment:
I made this an array of integers so that we could store the ids of the memories that correspond to each media type – e.g. if a memory is a video, we will store its id in the video array. 

When we initially talked last week you mentioned database 'normalisation' I think, is this not the best way to go about it?